### PR TITLE
dialects: add riscv.li operation

### PR DESCRIPTION
**Pull request by nazavode**

### CODE REVIEW

File: xdsl/dialects/riscv.py [hunk 1]
```diff
@@ -72,6 +74,32 @@ def __init__(
         )
 
 
+class RdImmOp(IRDLOperation, ABC):
```
Comment (superlopuh) [inline xdsl/dialects/riscv.py:77]:
```suggestion
class RdImmOperation(IRDLOperation, ABC):
```

File: tests/filecheck/dialects/riscv/riscv_ops.mlir [hunk 2]
```diff
@@ -6,4 +6,6 @@
   // CHECK: %2 = "riscv.add"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
   %3 = "riscv.sub"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
   // CHECK-NEXT: %3 = "riscv.sub"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %4 = "riscv.li"() {"immediate" = 1 : i32}: () -> !riscv.reg<>
```
Comment (superlopuh) [inline tests/filecheck/dialects/riscv/riscv_ops.mlir:9]: maybe to decouple the order of things, we could change the ssa value names back? so here would be something like this:
```suggestion
  %li = "riscv.li"() {"immediate" = 1 : i32}: () -> !riscv.reg<>
```

File: xdsl/dialects/riscv.py [hunk 3]
```diff
@@ -72,6 +74,32 @@ def __init__(
...
+    """
+    A base class for RISC-V operations that have one destination register, and one
+    immediate operand.
+
+    This is an assembler pseudo-instruction.
```
Comment (AntonLydike) [inline xdsl/dialects/riscv.py:82]: Not all RdImmOp are actually pseudo instructions, no?

File: xdsl/dialects/riscv.py [hunk 4]
```diff
@@ -100,10 +128,22 @@ class SubOp(RdRsRsOp):
     name = "riscv.sub"
 
 
+@irdl_op_definition
+class LiOp(RdImmOp):
```
Comment (AntonLydike) [inline xdsl/dialects/riscv.py:132]: Can we add something here to mark it as a pseudo op?

**Review by AntonLydike — APPROVED** [review]
Minor nitpicks. Very exciting to see this pick up steam!

File: xdsl/dialects/riscv.py (hunk 4 above)
Comment (math-fehr) [reply xdsl/dialects/riscv.py:128]: We can probably use traits for that?

**Review by math-fehr — APPROVED** [review]
:+1:

**Pull request merged**